app/src/main/java/com/github/bgabriel998/softwaredevproject/GeonamesHandler.java

```diff
@@ -1,99 +1,235 @@
-     * @param poiMaxResult max number of POI returned
-     * @param poiMaxRange max range for the POI request in km
-     * @throws InterruptedException interrupt exception occurs if thread is interrupted
+     * Search for POI.
+     * @param url full URL request, built with #urlForPOISearch or equivalent.
+     * Main requirements: <br>
+     * - Content must be in JSON format<br>
+     * - ways and relations must contain the "center" element. <br>
+     * @return elements as a list of POI
      */
-    public void startGetSurroundingPeaks(GeoPoint userLocation,int poiMaxResult, int poiMaxRange){
-        queryThread = new Thread(new Runnable() {
-            @Override
-            public void run() {
-                try{
-                    POIs = getPOI(userLocation, poiMaxResult, poiMaxRange);
+    private ArrayList<POI> getPOIsFromUrl(String url){
+        Log.d(BonusPackHelper.LOG_TAG, "OverpassAPIProvider:getPOIsFromUrl:"+url);
+        String jString = BonusPackHelper.requestStringFromUrl(url);
+
+        try {
+            //parse JSON and build POIs
+            JsonParser parser = new JsonParser();
+            JsonElement json = parser.parse(jString);
+            JsonObject jResult = json.getAsJsonObject();
+            JsonArray jElements = jResult.get("elements").getAsJsonArray();
+            ArrayList<POI> pois = new ArrayList<POI>(jElements.size());
+            for (JsonElement j:jElements){
+                JsonObject jo = j.getAsJsonObject();
+                POI poi = new POI(POI.POI_SERVICE_OVERPASS_API);
+                poi.mId = jo.get("id").getAsLong();
+                poi.mCategory = jo.get("type").getAsString();
+                if (jo.has("tags")){
+                    JsonObject jTags = jo.get("tags").getAsJsonObject();
+                    poi.mType = tagValueFromJson("name", jTags);
+                    //Try to set a relevant POI type by searching for an OSM commonly used tag key, and getting its value:
+                    poi.mDescription = tagValueFromJsonNotNull("amenity", jTags)
+                            + tagValueFromJsonNotNull("boundary", jTags)
+                            + tagValueFromJsonNotNull("building", jTags)
+                            + tagValueFromJsonNotNull("craft", jTags)
+                            + tagValueFromJsonNotNull("emergency", jTags)
+                            + tagValueFromJsonNotNull("highway", jTags)
+                            + tagValueFromJsonNotNull("historic", jTags)
+                            + tagValueFromJsonNotNull("landuse", jTags)
+                            + tagValueFromJsonNotNull("leisure", jTags)
+                            + tagValueFromJsonNotNull("natural", jTags)
+                            + tagValueFromJsonNotNull("shop", jTags)
+                            + tagValueFromJsonNotNull("sport", jTags)
+                            + tagValueFromJsonNotNull("tourism", jTags);
```

Comment:
Also here We could split this up to some set description function

Reply:
This function is copied directly from the provider and has to stay unchanged to avoid unexpected behavior